test/python/quantum_info/operators/symplectic/test_stabilizer_table.py

```diff
@@ -48,92 +51,97 @@ class TestStabilizerTableInit(QiskitTestCase):
 
     def test_array_init(self):
         """Test array initialization."""
-
-        with self.subTest(msg="bool array"):
-            target = np.array([[False, False], [True, True]])
-            value = StabilizerTable(target)._array
-            self.assertTrue(np.all(value == target))
-
-        with self.subTest(msg="bool array no copy"):
-            target = np.array([[False, True], [True, True]])
-            value = StabilizerTable(target)._array
-            value[0, 0] = not value[0, 0]
-            self.assertTrue(np.all(value == target))
-
-        with self.subTest(msg="bool array raises"):
-            array = np.array([[False, False, False], [True, True, True]])
-            self.assertRaises(QiskitError, StabilizerTable, array)
+        with self.assertWarns(DeprecationWarning):
```

Comment:
Recurring all the way through this file: `self.assertWarns` is an assertion, so it should be applied separately to each line that should raise a warning (and _only_ those lines).